Code/Source/RobotImporter/URDF/VisualsMaker.cpp

```diff
@@ -0,0 +1,208 @@
+                AZ::Vector3 scaleVector = URDF::TypeConversions::ConvertVector3(meshGeometry->scale);
+                bool isUniformScale = AZ::IsClose(scaleVector.GetMaxElement(), scaleVector.GetMinElement(), AZ::Constants::FloatEpsilon);
+                if (!isUniformScale)
+                {
+                    entity->CreateComponent<AzToolsFramework::Components::EditorNonUniformScaleComponent>();
+                }
+
+                entity->Activate();
+                // Set asset path
+                AZ::Render::MeshComponentRequestBus::Event(
+                    entityId, &AZ::Render::MeshComponentRequestBus::Events::SetModelAssetPath, assetPath.c_str());
+                // Set scale, uniform or non-uniform
+                if (isUniformScale)
+                {
+                    AZ::TransformBus::Event(entityId, &AZ::TransformBus::Events::SetLocalUniformScale, scaleVector.GetX());
+                }
+                else
+                {
+                    AZ::NonUniformScaleRequestBus::Event(entityId, &AZ::NonUniformScaleRequests::SetScale, scaleVector);
+                }
+                entity->Deactivate();
+            }
+            break;
+        default:
+            AZ_Warning("AddVisual", false, "Unsupported visual geometry type, %d", geometry->type);
+            return;
+        }
+    }
+
+    void VisualsMaker::AddMaterialForVisual(urdf::VisualSharedPtr visual, AZ::EntityId entityId)
+    {
+        // URDF does not include information from <gazebo> tags with specific materials, diffuse, specular and emissive params
+        if (!visual->material || !visual->geometry)
+        {
+            // Material is optional, and it requires geometry
+            return;
+        }
+
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+
+        // If present in map, take map color definition as priority, otherwise apply local node definition
+        urdf::Color materialColorUrdf = m_materials.find(visual->material->name) == m_materials.end()
+            ? visual->material->color
+            : m_materials[visual->material->name]->color;
+
+        AZ::Color materialColor = URDF::TypeConversions::ConvertColor(materialColorUrdf);
+        bool isPrimitive = visual->geometry->type != urdf::Geometry::MESH;
+        if (isPrimitive)
+        { // For primitives, set the color in the shape component
+            entity->Activate();
+            LmbrCentral::EditorShapeComponentRequestsBus::Event(
+                entityId, &LmbrCentral::EditorShapeComponentRequests::SetShapeColor, materialColor);
+            entity->Deactivate();
+            return;
+        }
+
+        // Mesh visual - we can have either filename or default material with a given color
+        // TODO - handle texture_filename - file materials
+        entity->CreateComponent(AZ::Render::EditorMaterialComponentTypeId);
+        AZ_Warning("AddVisual", false, "Setting color for material %s", visual->material->name.c_str());
```

Comment:
Is this really a warning or a just a regular info log?

For a regular information log, `AZ_TracePrintf` should be used with a newline at the end.